src/main/java/com/amazon/opendistroforelasticsearch/security/filter/OpenDistroSecurityFilter.java

```diff
@@ -291,7 +292,11 @@ public int order() {
             if (log.isDebugEnabled()) {
                 log.debug(pres);
             }
-            
+
+            if(threadContext.getTransient(OPENDISTRO_SECURITY_USER_ROLES_STRING) == null) {
```

Comment:
nit: populate only if there is a call to `chain`. If access is denied, it is necessary to populate `OPENDISTRO_SECURITY_USER_ROLES_STRING`?

Reply:
Does this require order of plugins loading to be enforced? If yes, how that will be done?

Reply:
Discussed with @skkosuri-amzn, the ordering needs to be researched. Can be done as a follow up PR.